src/d/actor/d_a_obj_ajav.cpp

```diff
@@ -7,185 +7,870 @@
+        /* SrcGObjAt Mtrl    */ 0,
+        /* SrcGObjAt SPrm    */ 0,
+        /* SrcGObjTg Se      */ 0,
+        /* SrcGObjTg HitMark */ 0,
+        /* SrcGObjTg Spl     */ 0,
+        /* SrcGObjTg Mtrl    */ 0,
+        /* SrcGObjTg SPrm    */ dCcG_TgSPrm_NoHitMark_e | dCcG_TgSPrm_NoConHit_e,
+        /* SrcGObjCo SPrm    */ 0
+    },
+    // cM3dGCylS
+    {{
+        /* Center */ {0.0f, -40.0f, 0.0f},
+        /* Radius */ 780.0f,
+        /* Height */ 3300.0f,
+    }},
+};
+
+static f32 l_daObjAjav_hint_cyl_h_talbe[] = {3300.0f, 2360.0f, 1050.0f};
+
+static f32 l_daObjAjav_cyl_r[6] = {
+    629.0f, 294.8f, 491.0f,
+    216.0f, 563.0f,  85.3f
+};
+
+static f32 l_daObjAjav_cyl_h[6] = {
+    773.0f, 458.5f, 1100.5f,
+    1401.9f, 1133.0f,  451.9
+};
+
+static const s32 l_daObjAjav_idx_table[6] = {
+    4, 5 ,6 ,7, 8, 9
+};
+
+static const char* const l_daObjAjav_ev_name[] = {
+    "ajav_destroy0",
+    "ajav_destroy1",
+    "ajav_uzu"
+};
 
 /* 000000EC-00000170       .text daObjAjav_make_splash__9daObjAjavF4cXyzf */
-void daObjAjav::daObjAjav_make_splash(cXyz, float) {
-    /* Nonmatching */
+void daObjAjav::daObjAjav_make_splash(cXyz i_splashPos, f32 i_scale) {
+    cXyz scale;
+
+    i_scale /= 2000.0f;
+    i_scale = 4.0f + i_scale;
+
+    scale.x = i_scale;
+    scale.y = 4.0f;
+    scale.z = scale.x;
+    dComIfGp_particle_set(dPa_name::ID_COMMON_003C, &i_splashPos,  NULL, &scale);
 }
 
 /* 000001AC-00000268       .text make_hamon__Q29daObjAjav6Part_cF4cXyzf */
-void daObjAjav::Part_c::make_hamon(cXyz, float) {
-    /* Nonmatching */
+void daObjAjav::Part_c::make_hamon(cXyz i_hamonPos, f32 i_scale) {
+    cXyz scale;
+
+    i_scale /= 2000.0f;
+    scale.x = 2.0f + i_scale;
+    scale.y = scale.x;
+    scale.z = scale.x;
+
+    i_hamonPos.y += 10.0f;
+    
+    f32 water_height;
+    if (fopAcM_getWaterY(&i_hamonPos, &water_height) == TRUE) {
+        i_hamonPos.y = water_height;
+    }
+
+    dComIfGp_particle_set(dPa_name::ID_COMMON_003F, &i_hamonPos, NULL, &scale);
 }
 
 /* 00000268-0000026C       .text no_proc__Q29daObjAjav6Part_cFPQ29daObjAjav5Act_c */
 void daObjAjav::Part_c::no_proc(daObjAjav::Act_c*) {
-    /* Nonmatching */
+    return;
 }
 
 /* 0000026C-000003A8       .text init_data__Q29daObjAjav6Part_cF4cXyz4cXyzP12dKy_tevstr_cP4cXyz */
-void daObjAjav::Part_c::init_data(cXyz, cXyz, dKy_tevstr_c*, cXyz*) {
-    /* Nonmatching */
+void daObjAjav::Part_c::init_data(cXyz i_actorPos, cXyz i_rockOffset, dKy_tevstr_c* i_tevStr, cXyz* i_rockPos) {
+    mCenterPos = i_rockOffset;
+    mCenterToOriginTrans = cXyz::Zero - mCenterPos;
+    mAccumulatedRotation = csXyz::Zero;
+    mRockDisplacement = cXyz::Zero;
+    mRockDisplacementRate = cXyz::Zero;
+    mpTevStr = i_tevStr;
+    mRockParticlePos = *i_rockPos + i_actorPos;
+#if VERSION > VERSION_DEMO
+    setExeProc(&Part_c::no_proc);
+    setDrawProc(&Part_c::draw_normal);
+#else
+    mExeProc = &Part_c::no_proc;
+    mDrawProc = &Part_c::draw_normal;
+#endif
```

Comment:
The version difference looks fake, does it need to be different to match both?
```suggestion
    setExeProc(&Part_c::no_proc);
    setDrawProc(&Part_c::draw_normal);
```

Reply:
using `setExeProc` and `setDrawProc` didn't match the demo in those cases, but looks like setting it directly matches both